### PR TITLE
Form Snippet for API Docs

### PR DESCRIPTION
**Pull request by Piinks**
## Description

This PR adds a code snippet for the Form class in he API docs. It also adds references to 'See also...' for other relevant widgets.

## Related Issues

Ref: #21136 

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes tests for *all* changed/updated/fixed behaviors (See [Test Coverage]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/lib/src/widgets/form.dart [hunk 1]
```diff
@@ -15,6 +15,56 @@ import 'will_pop_scope.dart';
 /// descendant of this [Form]. To obtain the [FormState], you may use [Form.of]
 /// with a context whose ancestor is the [Form], or pass a [GlobalKey] to the
 /// [Form] constructor and call [GlobalKey.currentState].
+///
+/// {@tool snippet --template=stateful_widget_material}
+/// This example shows a [Form] with one [TextFormField] and a [RaisedButton]. A
+/// [GlobalKey] is required to identify the [Form], and it is used here to
```
Comment (goderbauer) [inline packages/flutter/lib/src/widgets/form.dart]: this sounds like you always have to provide a global key, but we’re only giving it one her so we can access the forms state for validation purposes, right?

File: packages/flutter/lib/src/widgets/form.dart [hunk 2]
```diff
@@ -15,6 +15,56 @@ import 'will_pop_scope.dart';
...
+/// Widget build(BuildContext context) {
+///   // Build a Form widget using the _formKey we created above
+///   return Form(
+///     key: _formKey,
+///     child: Column(
+///       crossAxisAlignment: CrossAxisAlignment.start,
+///       children: <Widget>[
+///         TextFormField(
+///         validator: (value) {
+///         if (value.isEmpty) {
```
Comment (goderbauer) [inline packages/flutter/lib/src/widgets/form.dart]: Looks like something is messed up with the indentation here.

File: packages/flutter/lib/src/widgets/form.dart [hunk 3]
```diff
@@ -15,6 +15,56 @@ import 'will_pop_scope.dart';
...
+/// This example shows a [Form] with one [TextFormField] and a [RaisedButton]. A
+/// [GlobalKey] is required to identify the [Form], and it is used here to
+/// validate input.
+///
+/// ```dart
+/// final _formKey = GlobalKey<FormState>();
+///
+/// @override
+/// Widget build(BuildContext context) {
+///   // Build a Form widget using the _formKey we created above
```
Comment (goderbauer) [inline packages/flutter/lib/src/widgets/form.dart]: I’d probably remove this comment. It doesn’t seem super-helpful...

File: packages/flutter/lib/src/widgets/form.dart [hunk 4]
```diff
@@ -15,6 +15,56 @@ import 'will_pop_scope.dart';
...
+///           ),
+///         ),
+///       ],
+///     ),
+///   );
+/// }
+/// ```
+/// {@end-tool}
+///
+/// See also:
```
Comment (goderbauer) [inline packages/flutter/lib/src/widgets/form.dart]: Nit: add an empty line after see also

File: packages/flutter/lib/src/widgets/form.dart [hunk 5]
```diff
@@ -15,6 +15,56 @@ import 'will_pop_scope.dart';
...
+///  * [GlobalKey], a key that is unique across the entire app.
+///  * [FormField], a single form field widget that maintains the current state.
+///  * [TextFormField], a convenience widget that wraps a [TextField] widget in
+///  a [FormField].
```
Comment (goderbauer) [inline packages/flutter/lib/src/widgets/form.dart]: Nit: align with [TextFormField] on the previous line.

File: packages/flutter/lib/src/widgets/form.dart [hunk 6]
```diff
@@ -15,6 +15,56 @@ import 'will_pop_scope.dart';
...
+///
```
Comment (goderbauer) [inline packages/flutter/lib/src/widgets/form.dart]: Nit: remove extra blank line

File: packages/flutter/lib/src/widgets/form.dart (hunk 1 above)
Comment (Piinks) [reply packages/flutter/lib/src/widgets/form.dart]: Yeah, as I understand it, the key is always required. Using it to validate the form is just an example. Have I misunderstood?

**Review by goderbauer — APPROVED** [review]
LGTM

**Comment by Piinks** [comment]
docs and tool_tests-windows are both green when checked on Cirrus, they do not seem to be reporting back the results. Merging with approval and checks. 👍

**Pull request merged**